Specs/Scene/Instanced3DModel3DTileContentProviderSpec.js

```diff
@@ -0,0 +1,176 @@
+    });
+
+    afterEach(function() {
+        scene.primitives.removeAll();
+    });
+
+    function verifyRender(tileset) {
+        tileset.show = false;
+        expect(scene.renderForSpecs()).toEqual([0, 0, 0, 255]);
+        tileset.show = true;
+        var pixelColor = scene.renderForSpecs();
+        expect(pixelColor).not.toEqual([0, 0, 0, 255]);
+        return pixelColor;
+    }
+
+    function verifyRenderBlank(tileset) {
+        tileset.show = false;
+        expect(scene.renderForSpecs()).toEqual([0, 0, 0, 255]);
+        tileset.show = true;
+        expect(scene.renderForSpecs()).toEqual([0, 0, 0, 255]);
+    }
+
+    function verifyRenderTileset(tileset) {
+        // Verify render before being picked
+        verifyRender(tileset);
+
+        // Change the color of the picked instance to yellow
+        var picked = scene.pickForSpecs();
+        expect(picked).toBeDefined();
+        picked.color = Color.clone(Color.YELLOW, picked.color);
+
+        // Expect the pixel color to be some shade of yellow
+        var pixelColor = verifyRender(tileset);
+        expect(pixelColor[0]).toBeGreaterThan(0);
+        expect(pixelColor[1]).toBeGreaterThan(0);
+        expect(pixelColor[2]).toEqual(0);
+        expect(pixelColor[3]).toEqual(255);
+
+        // Turn show off and on
+        picked.show = false;
+        verifyRenderBlank(tileset);
+        picked.show = true;
+        verifyRender(tileset);
+    }
+
+    function loadTileset(url) {
+        var tileset = scene.primitives.add(new Cesium3DTileset({
+            url : url
+        }));
+
+        return pollToPromise(function() {
+            // Render scene to progressively load the content
+            scene.renderForSpecs();
+            return tileset.ready && (tileset._root.isReady());
+        }).then(function() {
+            return tileset;
+        });
+    }
+
+    function loadTileExpectProcessing(url) {
+        var tileset = {};
+        var tile = {};
+        var instancedTile = new Instanced3DModel3DTileContentProvider(tileset, tile, url);
+        return loadArrayBuffer(url).then(function(arrayBuffer) {
+            instancedTile.initialize(arrayBuffer);
+            // Expect to stay in the processing state
+            for (var i = 0; i < 10; ++i) {
+                instancedTile.update(tileset, scene.frameState);
+                expect(instancedTile.state).toEqual(Cesium3DTileContentState.PROCESSING);
+            }
+        });
+    }
+
+    function loadTileExpectError(url) {
+        var tileset = {};
+        var tile = {};
+        var instancedTile = new Instanced3DModel3DTileContentProvider(tileset, tile, url);
+        return loadArrayBuffer(url).then(function(arrayBuffer) {
+            expect(function() {
+                instancedTile.initialize(arrayBuffer);
+                instancedTile.update(tileset, scene.frameState);
+            }).toThrowDeveloperError();
+        });
+    }
+
+    it('throws with invalid magic', function() {
```

Comment:
Instead of having sample data on disk for all the failure cases, is it better to procedurally create the arraybuffer in the test?